core/src/main/scala/kafka/utils/Json.scala

```diff
@@ -74,8 +75,27 @@ object Json {
    *   T => null | Boolean | String | Number | Map[String, T] | Array[T] | Iterable[T]
    * Any other type will result in an exception.
    *
-   * This method does not properly handle non-ascii characters.
+   * This method does not properly handle non-ascii characters hence it is getting replaced by Json.encodeAsBytes
    */
-  def encodeAsBytes(obj: Any): Array[Byte] = encode(obj).getBytes(StandardCharsets.UTF_8)
+  def legacyEncodeAsBytes(obj: Any): Array[Byte] = legacyEncodeAsString(obj).getBytes(StandardCharsets.UTF_8)
```

Comment:
Is this still used by anything?

Reply:
Thanks for pointing as I didn't realize this. Its not getting used anywhere so removed this.